types/wavesurfer.js/types/backend.d.ts

```diff
@@ -0,0 +1,70 @@
+import MediaElement from "../src/mediaelement";
+import MediaElementWebAudio from "../src/mediaelement-webaudio";
+import Observer from "../src/util/observer";
+import WebAudio from "../src/webaudio";
+import { WaveSurferParams } from "./params";
+
+export type WaveSurferBackend = WebAudio | MediaElement | MediaElementWebAudio;
+
+export type Peaks = ReadonlyArray<number> | ReadonlyArray<ReadonlyArray<number>>;
+
+export abstract class WebAudioBackend extends Observer {
+    constructor(params: WaveSurferParams);
+
+    static scriptBufferSize: number;
+
+    /** Create analyser node to perform audio analysis. */
+    createAnalyserNode(): void;
+    /** Create ScriptProcessorNode to process audio. */
+    createScriptNode(): void;
+    /** Create the gain node needed to control the playback volume. */
+    createVolumeNode(): void;
+    /** This is called when wavesurfer is destroyed */
+    destroy(): void;
+    /** Destroy all references with WebAudio, disconnecting audio nodes and closing Audio Context. */
+    destroyWebAudio(): void;
+    /** Get the audio context used by this backend or create one. */
+    getAudioContext(): AudioContext;
+    /** Returns the current time in seconds relative to the audio-clip's duration. */
+    getCurrentTime(): number;
```

Comment:
This is OK and doesn't need to be changed by the way, unlike the MediaElement backend this always returns `number` 👍

Reply:
If I understand this correctly, `MediaElement.getCurrentTime()` (and `MediaElementWebAudio.getCurrentTime()`) returns `number | undefined`, whereas `WebAudio.getCurrentTime()` always returns `number`.

Reply:
Fixed in be2d8d6.